File: Examples/CSharp/AsposePdfFacades/Annotations/ImportAnnotations.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;
using System;
using Aspose.Pdf.InteractiveFeatures.Annotations;

namespace CSharp.AsposePDFFacades.Annotations
{
    public class ImportAnnotations
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Annotations();
            //create PdfAnnotationEditor object
            PdfAnnotationEditor AnnotationEditor = new PdfAnnotationEditor();
            //open PDF document
            AnnotationEditor.BindPdf(dataDir+ "ImportAnnotations.pdf");
            //import annotations
            FileStream fileStream = new System.IO.FileStream(dataDir+ "annotations.xfdf", System.IO.FileMode.Open, System.IO.FileAccess.Read);
            Enum[] annotType = { AnnotationType.FreeText, AnnotationType.Line };
            AnnotationEditor.ImportAnnotationFromXfdf(fileStream, annotType);
            //save output PDF
            AnnotationEditor.Save(dataDir + "ImportAnnotations_out.pdf");
 
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Annotations/ImportAnnotations.vb
Imports System.IO
Imports System
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades
Imports Aspose.Pdf.Annotations
Namespace AsposePDFFacades.Annotations
    Public Class ImportAnnotations
        Public Shared Sub Run()
            ' ExStart:ImportAnnotations
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Annotations()
            ' Create PdfAnnotationEditor object
            Dim AnnotationEditor As New PdfAnnotationEditor()
            ' Open PDF document
            AnnotationEditor.BindPdf(dataDir & Convert.ToString("ImportAnnotations.pdf"))
            ' Import annotations
            Dim fileStream As FileStream = New System.IO.FileStream(dataDir & Convert.ToString("annotations.xfdf"), System.IO.FileMode.Open, System.IO.FileAccess.Read)
            Dim annotType As [Enum]() = {AnnotationType.FreeText, AnnotationType.Line}
            AnnotationEditor.ImportAnnotationFromXfdf(fileStream, annotType)
            ' Save output PDF
            AnnotationEditor.Save(dataDir & Convert.ToString("ImportAnnotations_out_.pdf"))
            ' ExEnd:ImportAnnotations            
        End Sub
    End Class
End Namespace